group/views.py

```diff
@@ -1,22 +1,150 @@
+        try:
+            group = Group.objects.get(id=group_id)
+            if request.user != group.creator:
+                raise PermissionDenied("Access Denied")
+            return group
+        except Group.DoesNotExist as exc:
+            raise Http404 from exc
+
+    def get(self, request, *args, **kwargs):
+        """Retrieve a list of all the groups"""
+        queryset = self.get_queryset()
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.serializer_class(page, many=True)
+            return self.get_paginated_response(serializer.data)
+        serializer = self.serializer_class(queryset, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def post(self, request):
+        """Create a group"""
         data = request.data.copy()
-        data['creator'] = request.user.id
+        user_id = request.user.id
+        data['creator'] = user_id
+        data['members'] = [user_id]
         serializer = GroupSerializer(data=data)
         if serializer.is_valid():
             serializer.save()
             return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
-    return None
+
+    def patch(self, request):
+        """Update members list"""
+        data = request.data
+        group_id = data.get("group_id")
+        members_ids = data.get("members")
+
+        data.pop('creator', None)
+        data.pop('name', None)
+        try:
+            group = self.get_object(request, group_id)
+            errors = []
+            for member in members_ids:
+                try:
+                    user = User.objects.get(id=member)
+                    if user.id not in group.members.all()\
+                            .values_list('id', flat=True):
+                        group.members.add(user)
+                except User.DoesNotExist:
+                    errors.append(f"User with ID {member} is not found")
+            serializer = GroupSerializer(group, data=data, partial=True)
+            if serializer.is_valid():
+                serializer.save()
+                return Response(serializer.data, status=status.HTTP_200_OK)
+            if errors:
+                return Response({"error": errors},
+                                status=status.HTTP_400_BAD_REQUEST)
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
+        except PermissionDenied:
+            return Response({"error": "Only group creator can add members"},
+                            status=status.HTTP_403_FORBIDDEN)
+        except Http404:
+            return Response({"error": "Group not found"},
+                            status=status.HTTP_404_NOT_FOUND)
+
+    def delete(self, request, group_id):
+        """Delete a group"""
+        try:
+            group = self.get_object(request, group_id)
```

Comment:
It looks like anyone that's authenticated can delete a group? Make sure it's only the creator deleting a group

Reply:
Please check again because I put in a check to ensure that only the group creator can delete

Reply:
Atleast I don't see that in the code, please point me to it

Reply:
I see it in another class the GroupDetailView, not this one